### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by lyubomir-sumup**
* Improve documentation
* Code optimizations
* Introduce custom service

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,186 +24,26 @@ try {
         'code' => 'YOUR-AUTHORIZATION-CODE'
     ]);
     $checkoutService = $sumup->getCheckoutService();
-    $checkoutResponse = $checkoutService->create(/* pass here the required arguments */);
+    $checkoutResponse = $checkoutService->create(/* . . . */);
```
Comment (haltsir) [inline README.md]: For me it's unclear what this comment means.

File: README.md [hunk 2]
```diff
@@ -24,186 +24,26 @@ try {
...
-|`\SumUp\Exceptions\SumUpServerException`   	 | This exception is thrown when there are http errors of [5xx](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_Server_errors). 	|
-|`\SumUp\Exceptions\SumUpConfigurationException` | This exception is thrown when you provide a bad configuration for initialization of the `\SumUp\SumUp` object.	|
-|`\SumUp\Exceptions\SumUpArgumentException`  	 | This exception is thrown when you don't provide a mandatory argument to a function. 	|
-|`\SumUp\Exceptions\SumUpSDKException`           | This is the main exception which others inherit from. So this is the last exception to handle in your code if you want to catch many exceptions.|
+* [How to authorize?](https://github.com/sumup/sumup-ecom-php-sdk/blob/master/docs/HowToAuthorize.md)
+* [How to handle exceptions?](https://github.com/sumup/sumup-ecom-php-sdk/blob/master/docs/ExceptionsHandling.md)
+* [How to use my own http client?](https://github.com/sumup/sumup-ecom-php-sdk/blob/master/docs/HowToOverrideHttpClient.md)
```
Comment (haltsir) [inline README.md]: It's HTTP :P

File: docs/ExceptionsHandling.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Exceptions handling
+
+Exceptions handling is an important part of our code. We pay attention to this detail and **we recommend to wrap every statement from this SDK with a `try {} catch() {}` clause**.
+
+You should at least handle `\SumUp\Exceptions\SumUpSDKException` exception but if you want you can handle all sorts of exceptions.
+
+```php
+try {
+    $sumup = new \SumUp\SumUp(/* configuration */);
```
Comment (haltsir) [inline docs/ExceptionsHandling.md]: Is it possible to use the constructor signature here?

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+| [\SumUp\Services\Transactions](https://github.com/sumup/sumup-ecom-php-sdk/blob/master/docs/reference/Transactions.md) | The service that manages transactions. |
+
+<!-- ## HTTP Clients and Response
+
+| Class name | Description |
+|---	     |---	       |
+| [\SumUp\HttpClients\Response]() | The response object that every service returns. |
+| [\SumUp\HttpClients\SumUpCUrlClient]() | The HTTP client for managing cURL requests. |
+| [\SumUp\HttpClients\SumUpGuzzleHttpClient]() | The HTTP client for managing [Guzzle HTTP](https://packagist.org/packages/guzzlehttp/guzzle) requests. |
+| [\SumUp\HttpClients\HttpClientsFactory]() | The factory class that creates HTTP clients. | -->
```
Comment (haltsir) [inline docs/README.md]: Is this block forgotten?

Comment (lyubomir-sumup) [reply docs/README.md]: This is left here for future use

File: docs/ExceptionsHandling.md (hunk 3 above)
Comment (lyubomir-sumup) [reply docs/ExceptionsHandling.md]: I don't understand what you mean

**Pull request merged**